expr.go

```diff
@@ -641,27 +641,34 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
 			return nil
 		}
 
+		re := regexp.MustCompile("^([a-z]+)([AB][a-z]+)$")
+		submatch := re.FindStringSubmatch(e.target)
+		var compute func([]float64, []bool) float64
+		switch submatch[1] {
+		case "average":
+			compute = avgValue
+		case "current":
+			compute = currentValue
+		case "maximum":
+			compute = maxValue
+		case "minimum":
+			compute = minValue
+		}
 		var results []*metricData
 		for _, a := range args {
-			switch e.target {
-			case "averageAbove":
-				if avgValue(a.Values, a.IsAbsent) >= float64(n) {
+			value := compute(a.Values, a.IsAbsent)
+			switch submatch[2] {
+			case "Above":
+				if value >= float64(n) {
```

Comment:
We keep needing to convert n to a float64.  It probably makes sense to have it be a floating point number in the first place.  Change the declaration above for n to `getFloatArg` instead of `getIntArg`.